src/fireo/models/model.py

```diff
@@ -386,7 +406,12 @@ class User(Model):
         # pass the model instance if want change in it after save, fetch etc operations
         # otherwise it will return new model instance
         return self.__class__.collection.create(
-            self, transaction, batch, merge, no_return, **self._get_fields(changed_only=merge)
+            self,
+            transaction,
+            batch,
+            merge,
+            no_return,
+            **self._get_fields(ignore_unchanged=merge, ignore_default_none=True)
```

Comment:
I'm not sure about this `ignore_unchanged=merge`
Should `model.upsert()` override default values?